pkg/controller/livenessprobemapnodeprobe/probemapnodeprobe_pod_event_handler.go

```diff
@@ -0,0 +1,64 @@
+package livenessprobemapnodeprobe
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ handler.EventHandler = &enqueueRequestForPod{}
+
+type enqueueRequestForPod struct {
+	reader client.Reader
+}
+
+func (p *enqueueRequestForPod) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	obj, ok := evt.Object.(*v1.Pod)
+	if !ok {
+		return
+	}
+	p.queue(q, obj)
+	return
+}
+
+func (p *enqueueRequestForPod) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	obj, ok := evt.Object.(*v1.Pod)
+	if !ok {
+		return
+	}
+	p.queue(q, obj)
+	return
+}
+
+func (p *enqueueRequestForPod) Generic(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+	obj, ok := evt.Object.(*v1.Pod)
+	if !ok {
+		return
+	}
+	p.queue(q, obj)
+	return
+}
+
+func (p *enqueueRequestForPod) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	new, ok := evt.ObjectNew.(*v1.Pod)
+	if !ok {
+		return
+	}
+	p.queue(q, new)
```

Comment:
if pod liveness probe does not change, it seems not necessary to enqueue again

Reply:
done, details:
`
// check the livenessProbe config from new and old pod annotation whether is changed or not
	getNewPodConfig := getRawEnhancedLivenessProbeConfig(new)
	getOldPodConfig := getRawEnhancedLivenessProbeConfig(old)
	if !reflect.DeepEqual(getNewPodConfig, getOldPodConfig) {
		p.queue(q, new)
	}
`